graph/coloring/backtracking.go

```diff
@@ -0,0 +1,47 @@
+package coloring
+
+// ColoringUsingBacktracking will return the Color of each vertex and the
+// total number of different colors used, using backtracking
+func (g *Graph) ColoringUsingBacktracking() (map[int]Color, int) {
```

Comment:
Naming is not fluent, consider changing it to:

```suggestion
func (g *Graph) ColorUsingBacktracking() (map[int]Color, int) {
```

This naming would give an English friendly readability. For example,

```Go
g := colouring.Graph{...}

g.ColorUsingBacktracking()
...
```

Reply:
Yes, now I also see that. I'll fix that.